src/Command.php

```diff
@@ -463,7 +463,17 @@ public function quality(int $quality): self
 
         return $this;
     }
-
+    /**
+     * @see http://imagemagick.org/script/command-line-options.php#density
+     */
+    public function density(int $density): self
+    {
+        $densityAry = [];
+        $densityAry[] = '-density';
+        $densityAry[] = (string) $density;
+        array_splice($this->command,2,0,$densityAry); //density option only works when it's on second place in command (tested for pdf to image)
```

Comment:
Density is not supposed to be placed "second place in command", but "before any vector source is specified".

If you use ImageMagick to manipulate multiple files or intermediary files, you can specify the density for all files.

You say you "tested for pdf to image", but if you test "image to pdf", you'll realize that the density option must be put before the pdf filename, like this for instance: `convert my-file.jpg -density 300 my-file.pdf`.

Also, the density option must respect its specified format: `{width} | {width}x{height}`.

You can add a small function to the `Reference` class that will validate these rules.

Reply:
I tried to read Reference classes but didn't get so much idea how i will gonna add these options but i'll try.